File: Tests/DataStructures/Stack`1_Tests.cs
using System;
using Xunit;

namespace Langly.DataStructures {
	public class Stack1_Tests {
		[Theory]
		[InlineData(new Int32[] { }, new Int32[] { })]
		[InlineData(new Int32[] { 1, 2, 3, 4, 5 }, new Int32[] { 5, 4, 3, 2, 1 })]
		public void Stack(Int32[] expected, Int32[] values) {
			Stack<Int32> stack = new Stack<Int32>().Push(values);
			Assert.Equal(expected.Length, stack.Count);
			Assert.Equal(expected, stack);
			stack.Pop(expected.Length, out Int32[] vals);
			Assert.Equal(0, stack.Count);
			Assert.Equal(expected, vals);
		}
	}
}

File: Tests.VisualBasic/DataStructures/Stack`1_Tests.vb
Imports Xunit

Namespace DataStructures
	Public Class Stack1_Tests
		<Theory>
		<MemberData(NameOf(Stack1_Data.Stack), MemberType:=GetType(Stack1_Data))>
		Public Sub Stack(expected As Int32(), values As Int32())
			Dim stack = New Stack(Of Int32)().Write(values)
			Assert.Equal(expected.Length, stack.Count)
			Assert.Equal(expected, stack)
			Dim vals As ReadOnlyMemory(Of Int32)
			Dim unused = stack.Read(expected.Length, vals)
			Assert.Equal(0, stack.Count)
			Assert.Equal(expected, vals.ToArray())
		End Sub
	End Class
End Namespace
